types/mongoose/index.d.ts

```diff
@@ -3590,4 +3590,7 @@ declare module "mongoose" {
     /** The collection name */
     name: string;
   }
+
+  /* Helper type to extract a definition type from a Document type */
+  type DocumentDefinition<T> = T extends Document & infer D ? D & { _id: mongodb.ObjectId } : never;
```

Comment:
Sorry for the delay, but it took me a little while to figure out what `infer` would do in this context.  One snag to watch out for is the handling of proper subtypes.  Suppose you had
```ts
class Document2 extends Document {
    special: number;
}
```
If you passed `Document2 & { prop1: number}` as `T`, you'd get back `Document2 & { prop1: number}` as `D`, which doesn't seem to be what you want.

Reply:
You are right! I updated the implementation of this helper to use `Omit` instead.